### PR TITLE
Projects page

### LINKED ISSUE #3: Update website content
**Is your feature request related to a problem? Please describe.**
Before the website gets launched, the data needs to be updated, including:

- Stats
- Past project reports
- Study resources
- Term 2/24 team structure

**Describe the solution you'd like**
Hope we can get this done before the development finishes. Also, ask CR how to get the site launched onto the domain, i.e. tutors-in-action.org.

Last but not least, check and report typos or bugs.

**Describe alternatives you've considered**
N/A

**Additional context**
May ask other TIA comittee members to write the reports in markdown.

### PR DESCRIPTION
**Pull request by AARandom**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #3 

## Type of change

- Content update

## Additions

- upgrades to the project page (courtesy of @aidil-sekandar, thank you sm!!!)
- added specific folder for post images in the static folder (also, thank you aidil)
- added reports for Joy For Angels, Sculpt your Personal Merit, and SPM In Action

## Limitations

- replaced markdown tables with image of tables because multiple tables render all at once in a file

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested the website locally with my changes, and I am sure that it works well on different screen sizes

### CODE REVIEW

**Comment by AARandom** [comment]
also, apparently html tags dont render in markdown. Is this the default or is there someway to enable it?

File: assets/projects.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.project-filter {
```
Comment (jaredliw) [inline assets/projects.css:1]: same, bs classes pls

File: layouts/projects/single.html [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
-        <hr>
+        <div class="project-metadata font-monospace fw-normal">
+            <span class="d-flex align-items-center gap-2">
+                <svg xmlns="http://www.w3.org/2000/svg" width="1.3em" height="1.3em" viewBox="0 0 24 24">
+                    <g fill="none">
+                        <path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z" />
+                        <path fill="#343a40" d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-7zm-5-9a1 1 0 0 1 1 1v1h2a2 2 0 0 1 2 2v3H3V7a2 2 0 0 1 2-2h2V4a1 1 0 0 1 2 0v1h6V4a1 1 0 0 1 1-1" />
+                    </g>
+                </svg>
+                <time>{{ .Date.Format "Jan 2, 2006" }}</time>
```
Comment (jaredliw) [inline layouts/projects/single.html:17]:
```suggestion
                <time>{{ .Date.Format "2 Jan 2006" }}</time>
```
Inconsistant format. i think this is better, more malaysian way of expressing dates

**Review by jaredliw — CHANGES_REQUESTED** [review]
Great! I find this design to be an icing on the cake. It's color theme matches TIA and it also addresses the feedback from other committee members.

Butttt, I would like to comment on two things:
- the bootsrap classes. bs is used to avoid css. as u can see, i dun write much css (and i dun need to). We wanna avoid different sizes and margins everywhere.
- the filter feature. See comments for more.

Due to the 2nd reason, pls open an issue for big changes and discuss on the design. It will give everyone a better idea on what is going on, and to reduce reviewing back and forth.

**Comment by jaredliw** [comment]
I didnt check on md ya, i safely assume the grammar and spelling are all correct.

**Comment by jaredliw** [comment]
> also, apparently html tags dont render in markdown. Is this the default or is there someway to enable it?

U need to specify this explicitly in `config.toml`:

```toml
[markup.goldmark.renderer]
    unsafe = true
```

HTML is not safe, user can inject malicious `<script>` to change the behavior of the site. But since we have full control on everything on the website, it wont be an issue.

However, I wont recommend doing so, using **shortcodes** is a better idea so that the design is more under control and non-tech ppl can write the project report easily without worrying about HTML.

**Comment by AARandom** [comment]
Hi Jared. thanks for checking over everything and giving ur feedback! I'll try and work on those now since project proposals are over. Thank youuuu <3

As for the html tags, its mainly just to keep the report the same as the pdf report. I guess its fine since its just formatting but I'll give it a try later. Thanks for the suggestion !!

File: layouts/projects/list.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{{- define "main" -}}
+    {{ partial "nav.html" (dict "nav_white" 1) }}
+
+    <section style="padding-top: 6.5rem" class="mb-5">
+        <div class="container">
+            <h2 class="text-center text-primary fw-lighter">Our <span class="fw-bold">projects</span>.</h2>
+        </div>
+    </section>
+
+    {{/* Filter projects base on status (Ongoing/complete) - disabled since there arent many projects yet.
```
[the remark this answers is not in the document]
Comment (AARandom) [reply layouts/projects/list.html:13]: Can we open an issue for this after? Cuz right now there isnt many projects to display so I think we can work on it after publishing the website

[the remark this answers is not in the document]
Comment (jaredliw) [reply layouts/projects/list.html:13]: Yes, ofcoz. I understand that we are rushing to publish it soonest possible (update me plsss)
But for now pls keep this (and oso the commented code below) in a branch, or using `git stash`.

[the remark this answers is not in the document]
Comment (jaredliw) [reply layouts/projects/list.html:13]: Oh yeah, regarding this, I think done is better than perfect. We shall complete the `join us` part and get things published, we can discuss on the improvements later. We just hv a liiiittle bit more, help lynn out pls.

Lemme know how u think.

File: assets/projects-single.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+/* Small devices (landscape phones, 576px and up) */
+@media (min-width: 576px) {
+    .project-metadata {
+        display: flex;
+        gap: 1em;
+        flex-wrap: wrap;
+    }
+}
+
```
Comment (jaredliw) [inline assets/projects-single.css:19]: as far as i can tell, all of the above can be written using bs. May u justify these pls?

**Review by jaredliw — COMMENTED** [review]
@AARandom Here u are. thankss

File: assets/projects.css (hunk 1 above)
Comment (jaredliw) [reply assets/projects.css:2]: reopen for `border`, `background-color` and `color`

Comment (AARandom) [reply assets/projects.css:2]: Sorry, thought these were css variables and got scared to touch it last time. Moved them to their tags in list.html

File: layouts/projects/list.html [hunk 5]
```diff
@@ -0,0 +1,75 @@
+{{- define "main" -}}
+    {{ partial "nav.html" (dict "nav_white" 1) }}
+
+    <script src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha384-GNFwBvfVxBkLMJpYMOABq3c+d3KnQxudP/mGPkzpZSTYykLBNsZEnG2D9G/X/+7D" crossorigin="anonymous" async></script>
```
[the remark this answers is not in the document]
Comment (AARandom) [reply layouts/projects/list.html:4]: I did the source and checked that the files are there, but unsure about the target property

2024-08-22 21_19_35-config toml - tia - Visual Studio Code

File: layouts/projects/list.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (AARandom) [reply layouts/projects/list.html:13]: Yea sure can. I'll open an update post in the pr separately from this one. I'll check up on lynn after this

**Comment by AARandom** [comment]
Updates:

Cr and the directors want to review the website first before publishing, which should be after completing this pr (not sure about the donate button, no updates about that but checking up later).

So, rn we're targetting publishing it before the august general meeting (too late dy), might extend it to before the september meeting.

**Comment by jaredliw** [comment]
> Updates:
> 
> Cr and the directors want to review the website first before publishing, which should be after completing this pr (not sure about the donate button, no updates about that but checking up later).
> 
> So, rn we're targetting publishing it before the august general meeting (too late dy), might extend it to before the september meeting.

Understood. If so, i will try to do smth with it before ur tmr's meeting, merging these two PRs.
I think everthing is ok alr. U may just share this link and ask for the review (what i did previously). And for bout a week's time then we can launch it alr (31 aug or 1 sept is a good time right?). I would oso like to learn on how to manage the domain name plss

File: layouts/projects/list.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaredliw) [reply layouts/projects/list.html]: @AARandom Ok, I stash the commented code in #3

File: assets/projects-single.css (hunk 4 above)
Comment (jaredliw) [reply assets/projects-single.css]: @AARandom Ok understood. But u can do it this way:

image

File: layouts/projects/list.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (jaredliw) [reply layouts/projects/list.html]: @AARandom Okok, it works like this:

The things in `node_modules` are not included in the bundled html code. "Mount" here tells Hugo to copy the things in `source` to `target` (the `static` folder).

We don't use CDN links to reduce latency and faster initial load.

image

image
Click on my commits to see how i get it done.

**Review by jaredliw — APPROVED** [review]
Ok done. To speed up the process, I made the following changes directly:
- Webstorm auto reformat, incl. spaces, spelling suggestions.
- Replace more css by bs classes
- Stash the project filter code into #3 for further discussions
- Mount `masonry-layout`

Pls check my commits and raise any concerns if u hv any, then, u may merge this pr
Thanks!

**Comment by AARandom** [comment]
Thanks Jared for all the time and effort you have put into helping me! (definitely couldnt finish this without you)
Sorry for making you do all those last commits tho, feeling guilty but you have my deepest gratitude!!

**Comment by jaredliw** [comment]
> Thanks Jared for all the time and effort you have put into helping me! (definitely couldnt finish this without you) Sorry for making you do all those last commits tho, feeling guilty but you have my deepest gratitude!!

 U guys are the one who did the great work! Im just wrapping things up
Thanks for your patience and sry for being annoying xD

**Pull request merged**